python/ray/dashboard/modules/node/node_head.py

```diff
@@ -442,18 +442,15 @@ async def _update_node_stats(self):
         #       from another async task)
         current_stub_node_id_tuples = list(self._stubs.items())
 
-        if current_stub_node_id_tuples:
-            node_ids, _ = zip(*current_stub_node_id_tuples)
-        else:
-            node_ids = []
-
+        node_ids = []
         get_node_stats_tasks = []
 
-        for i, (node_id, stub) in enumerate(current_stub_node_id_tuples):
+        for _, (node_id, stub) in enumerate(current_stub_node_id_tuples):
             node_info = DataSource.nodes.get(node_id)
             if node_info["state"] != "ALIVE":
                 continue
 
+            node_ids.append(node_id)
```

Comment:
Before `node_ids` contains dead nodes but `get_node_stats_tasks` doesn't, hence the check failure.

Reply:
Let's add a regression test

Reply:
Added to `test_node.py`